### PR TITLE
Java Input Output 

### PR DESCRIPTION
**Pull request by dskrumov**

### CODE REVIEW

File: src/com/company/reverseddocument/ReadFile.java [hunk 1]
```diff
@@ -0,0 +1,12 @@
+package com.company.reverseddocument;
+
+
+import java.io.IOException;
+
+public class ReadFile {
```
Comment (mgenov) [inline src/com/company/reverseddocument/ReadFile.java]: ??

File: src/com/company/readmethods/ReadTypes.java [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+import java.util.Scanner;
+
+
+public class ReadTypes {
+
+    Scanner scanner = new Scanner(System.in);
+
+    public void readString(String data) {
+        System.out.print("Enter data: ");
```
Comment (mgenov) [inline src/com/company/readmethods/ReadTypes.java]: System.out should not be used in here. 

All methods here should return values. Here is an example signature: 
```java
public String readString() {
}
```

File: src/com/company/readmethods/ReadTypes.java [hunk 3]
```diff
@@ -0,0 +1,37 @@
+package com.company.readmethods;
+
+import java.util.Scanner;
+
+
+public class ReadTypes {
```
Comment (mgenov) [inline src/com/company/readmethods/ReadTypes.java]: Read is a verb not a noun. 

ConsoleReader  sounds better name.

File: src/com/company/readfromconsole/Main.java [hunk 4]
```diff
@@ -0,0 +1,15 @@
+package com.company.readfromconsole;
+import java.io.IOException;
+
+
+public class Main {
+    public static void main(String[] args) throws IOException {
+
+        ReadData readData = new ReadData();
+
+        readData.createFile("Test.txt");
```
Comment (mgenov) [inline src/com/company/readfromconsole/Main.java]: The flow for reading looks too complicated. 

new File("Test.txt").reverseTillReach(".") looks like a better api.

File: src/com/company/reverseddocument/ReadFile.java (hunk 1 above)
Comment (dskrumov) [inline src/com/company/reverseddocument/ReadFile.java]: Just started the 3rd exerc thats why its empty.

File: src/com/company/readmethods/ReadTypes.java (hunk 2 above)
Comment (dskrumov) [inline src/com/company/readmethods/ReadTypes.java]: So if the method should return value, it has to look like that:
public String readString() {
        String name = scanner.next();
        return name;
    }    

or am i wrong ?

Comment (mgenov) [inline src/com/company/readmethods/ReadTypes.java]: getX, readX,  findX should return values as the name says so.

File: src/com/company/readfromconsole/File.java [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.Scanner;
+
+public class File {
+
+    /**
+     * Fields
+     */
+    Scanner scanner = new Scanner(System.in);
```
Comment (martinmilev) [inline src/com/company/readfromconsole/File.java]: private

File: src/com/company/readfromconsole/File.java [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+
+    /**
+     * Write on the previous created file untill reach the separator
+     *
+     * @param separator is string, which will stop the scanner
+     * @throws IOException
+     */
+    public void writeTillReach(String separator) {
+        this.chachacter = separator;
+        try {
```
Comment (martinmilev) [inline src/com/company/readfromconsole/File.java]: I think there is nothing too catch here.

File: src/com/company/readmethods/ConsoleReader.java [hunk 7]
```diff
@@ -0,0 +1,45 @@
+package com.company.readmethods;
+
+import java.util.Scanner;
+
+
+public class ConsoleReader {
+
+    Scanner scanner = new Scanner(System.in);
```
Comment (martinmilev) [inline src/com/company/readmethods/ConsoleReader.java]: private

File: src/com/company/readmethods/ConsoleReader.java [hunk 8]
```diff
@@ -0,0 +1,45 @@
...
+
+    /**
+     * Read string from the console
+     * @return a string
+     */
+    public String readString() {
+        String name = scanner.next();
+        return name;
```
Comment (martinmilev) [inline src/com/company/readmethods/ConsoleReader.java]: you can `return scanner.next();`

File: src/com/company/readmethods/ConsoleReader.java [hunk 9]
```diff
@@ -0,0 +1,45 @@
...
+        int number = scanner.nextInt();
+        return number;
+    }
+
+    /**
+     * Read a char from the console
+     * @return the char
+     */
+    public char readChar() {
+        char ch = scanner.next().charAt(2);
```
Comment (martinmilev) [inline src/com/company/readmethods/ConsoleReader.java]: why `charAt(2);`?

File: src/com/company/readmethods/Main.java [hunk 10]
```diff
@@ -0,0 +1,18 @@
+package com.company.readmethods;
+import java.util.InputMismatchException;
+
+public class Main {
+    public static void main(String[] args) {
+        ConsoleReader readTypes = new ConsoleReader();
+        try {
```
Comment (martinmilev) [inline src/com/company/readmethods/Main.java]: why catching exceptions

File: src/com/company/reversedocument/File.java [hunk 11]
```diff
@@ -0,0 +1,46 @@
+package com.company.reversedocument;
+import java.io.*;
+public class File {
+    /**
+     * Fields
+     */
+    private String line;
+    private String temp = "";
```
Comment (martinmilev) [inline src/com/company/reversedocument/File.java]: don't need to declare "temp", "line" here.

File: src/com/company/streams/Main.java [hunk 12]
```diff
@@ -0,0 +1,14 @@
+package com.company.streams;
+
+import java.io.IOException;
+
+public class Main {
+    public static void main(String[] args) throws IOException {
+
+        byte[] array = {'1', '2', '3', '4', '5', '6','7', '8', '9','0'};
```
Comment (martinmilev) [inline src/com/company/streams/Main.java]: use files, not array

File: src/com/company/streams/ObjectTransfer.java [hunk 13]
```diff
@@ -0,0 +1,27 @@
...
+    private InputStream inputStream;
+    private OutputStream outputStream;
+
+    /**
+     * Read from the array which is defined in main method and writes the data with specific length
+     * and offset in the streamTEST.txt
+     * @param array is the defined by us array
+     * @throws IOException
+     */
+    public void ByteArrayInputStream(byte[] array, int offset, int length) throws IOException {
```
Comment (martinmilev) [inline src/com/company/streams/ObjectTransfer.java]: Methods names should start with lowercase, should not throw Exception here.

File: src/com/company/reversedocument/File.java [hunk 14]
```diff
@@ -0,0 +1,46 @@
...
+    private FileReader fileReader;
+    private BufferedReader bufferedReader;
+
+    /**
+     * Create the file
+     * @param documentReadName is the name of the created file
+     */
+    public File(String documentReadName) {
+        try {
```
Comment (martinmilev) [inline src/com/company/reversedocument/File.java]: Constructor should not contain this.

**Pull request closed without merging**